src/Analyzers/CSharp/Analyzers/UseImplicitObjectCreation/CSharpUseImplicitObjectCreationDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,122 @@
+
+#nullable enable
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Utilities;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseImplicitObjectCreation
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class UseImplicitObjectCreationDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        public UseImplicitObjectCreationDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.UseImplicitObjectCreationDiagnosticId,
+                   CSharpCodeStyleOptions.ImplicitObjectCreationWhenTypeIsApparent,
+                   LanguageNames.CSharp,
+                   new LocalizableResourceString(nameof(CSharpAnalyzersResources.Use_new), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)),
+                   new LocalizableResourceString(nameof(CSharpAnalyzersResources.new_expression_can_be_simplified), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterSyntaxNodeAction(AnalyzeSyntax, SyntaxKind.ObjectCreationExpression);
+
+        private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var options = context.Options;
+            var syntaxTree = context.Node.SyntaxTree;
+            var semanticModel = context.SemanticModel;
+            var cancellationToken = context.CancellationToken;
+
+            var optionSet = options.GetAnalyzerOptionSet(syntaxTree, cancellationToken);
+            var styleOption = options.GetOption(CSharpCodeStyleOptions.ImplicitObjectCreationWhenTypeIsApparent, syntaxTree, cancellationToken);
+            if (!styleOption.Value)
+            {
+                // Bail immediately if the user has disabled this feature.
+                return;
+            }
+
+            // type is apparent if we the object creation location is closely tied (spatially) to the explicit type.  Specifically:
+            //
+            // 1. Variable declarations.    i.e. `List<int> list = new ...`.  Note: we will suppress ourselves if this
+            //    is a field and the 'var' preferences would lead to preferring this as `var list = ...`
+            // 2. Expr-bodied constructs with an explicit return type.  i.e. `List<int> Prop => new ...` or `List<int>
+            //    GetX(...) => ...` The latter doesn't necessarily have the object creation spatially next to the type.
+            //    However, the type is always in a very easy to ascertain location in c#, so it counts as apparent. 
```

Comment:
```suggestion
            // 1. Variable declarations.    i.e. `T value = new T(...)`.  Note: we will suppress ourselves if this
            //    is a field and the 'var' preferences would lead to preferring this as `var value = new T(...)`
            // 2. Expression-bodied constructs with an explicit return type.  i.e. `T Property => new T(...)` or
            //    `T GetValue(...) => new T(...)` The latter doesn't necessarily have the object creation spatially next to the type.
            //    However, the type is always in a very easy to ascertain location in C#, so it is treated as apparent. 
```

Reply:
i'm ok using `List<int>` :)

Reply:
The change was more about being explicit with the right hand side, i.e. is `...` talking about `new()` or `new T()`. I used a shorter type so the longer right hand side didn't make it notably longer than the original.